Demo/Demo/ViewModels/CardPaymentViewModel.swift

```diff
@@ -0,0 +1,112 @@
+import Foundation
+import CardPayments
+import CorePayments
+
+class CardPaymentViewModel: ObservableObject, CardDelegate {
+
+    @Published var state = CardPaymentState()
+
+    func createOrder(amount: String, selectedMerchantIntegration: MerchantIntegration, intent: String) async throws {
+        // might need to pass in payee as payee object or as auth header
+
+        let amountRequest = Amount(currencyCode: "USD", value: amount)
+        // TODO: might need to pass in payee as payee object or as auth header
+        let orderRequestParams = CreateOrderParams(
+            intent: intent,
+            purchaseUnits: [PurchaseUnit(amount: amountRequest)]
+        )
+
+        do {
+            DispatchQueue.main.async {
+                self.state.createdOrderResponse = .loading
+            }
+            let order = try await DemoMerchantAPI.sharedService.createOrder(
+                orderParams: orderRequestParams, selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.createdOrderResponse = .loaded(order)
+                print("✅ fetched orderID: \(order.id) with status: \(order.status)")
+            }
+        } catch {
+            DispatchQueue.main.async {
+                self.state.createdOrderResponse = .error(message: error.localizedDescription)
+                print("❌ failed to fetch orderID: \(error)")
+            }
+        }
+    }
+
+    func checkoutWith(
+        card: Card,
+        orderID: String
+    ) async {
+        do {
+            DispatchQueue.main.async {
+                self.state.approveResultResponse = .loading
+            }
+            let config = try await getCoreConfig()
+            let cardClient = CardClient(config: config)
+            cardClient.delegate = self
+            // SCA from UI?
+            let cardRequest = CardRequest(orderID: orderID, card: card, sca: .scaAlways)
+            cardClient.approveOrder(request: cardRequest)
+        } catch {
+            self.state.approveResultResponse = .error(message: error.localizedDescription)
+            print("failed in checkout with card. \(error.localizedDescription)")
+        }
+    }
+
+    // Two functions below should be in base or core ViewModel that is
+    // inherited by feature ViewControllers
```

Comment:
Actually this comment needs I am deleting these in favor having static function in DemoSettings return clientID depending on selected environment and selected integration.

Reply:
I am going to put these functions in an EnvironmentObject. These are functions that simply return values, having them available from main featureSelectionView would be good so they can be accessed from any child views.